### ISSUE TITLE
Replace DeletedNavigationWarning in the inspector with a Notice

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#59785]
### Description

When a menu has been deleted but it's still set as the current menu of a Navigation block, a `DeletedNavigationWarning` which is a `Warning` component is shown, both in the editor canvas and in the block inspector. Screenshot:

![Screenshot 2024-03-12 at 13 51 02](https://github.com/WordPress/gutenberg/assets/1682452/3783b2e6-cd86-45a5-9873-65ae4cc93f07)

To me, while this warning design makes sense in the canvas, it's less than ideal in the inspector.

The Notice component would be more appropriate in the inspector:
- I may be missing something but I'm not sure there are other usages of a Warning component _in the inspector_ or at least I haven't seen it, so far.
- A Notice would be more consistent with other Notices shown in the inspector.
- A Notice handles accessibility by the means of an ARIA role, while a Warning component doesn't.

I'd like to propose to change it to a Notice. Screenshot:

![Screenshot 2024-03-12 at 14 08 16](https://github.com/WordPress/gutenberg/assets/1682452/8ad2711f-6458-47f4-a6fd-e7f949c9bd36)

### Step-by-step reproduction instructions

- Go to the site editor.
- Select a Navigation block that does have a menu.
- In the Navigation block inspector, go to the Settings tab, scroll down to 'Advanced', expand it and click 'Delete menu'.
- Go back to the Navigation block inspector > List view tab
- Obsrve the warning in the inspector

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
